config/provider/upgrade/config.go

```diff
@@ -0,0 +1,11 @@
+package providerupgradeconfig
+
+type Configurations struct {
```

Comment:
nitpick:
```suggestion
type Configuration struct {
```
Could also just call it `Config` potentially. Singular just makes a bit more sense to me as the struct contains _all_ "Configuration" and the embedded `Providers` are the "Configurations".